ccxt.d.ts

```diff
@@ -58,19 +58,33 @@ declare module 'ccxt' {
         quote: string;
         baseId: string;
         quoteId: string;
+        settle: string;
+        settleId?: string;
         type?: string;
         spot?: boolean;
         margin?: boolean;
         swap?: boolean;
         future?: boolean;
         option?: boolean;
         active: boolean;
+        contract: boolean;
+        linear: boolean;
+        inverse: boolean;
         precision: { base: number, quote: number, amount: number, price: number };
```

Comment:
Let's remove `base` and `quote`, cause they're exchange-specific, let's keep `amount: number | undefined` and `price: number | undefined`. We should also add the `undefined` variation to all `limits`.

Reply:
@kroitor regarding the limits, I've noticed we already have a MinMax type, so I used it